global-api/importer/ghgrp_epa/ghgrp_importer.py

```diff
@@ -0,0 +1,152 @@
+    return str(uuid.uuid3(namespace, name))
+
+def load_direct_emitters_from_zip(zip_file_path: str, file_name: str):
+    """Load 'Direct Emitters' sheet from a specific XLSX file inside a zip archive into a pandas DataFrame"""
+    with zipfile.ZipFile(zip_file_path, 'r') as zip_file:
+        # Check if the specified file is in XLSX format
+        if file_name.lower().endswith('.xlsx'):
+            with zip_file.open(file_name) as xlsx_file:
+                # Read the 'Direct Emitters' sheet from the XLSX file
+                df = pd.read_excel(xlsx_file, sheet_name='Direct Emitters')
+                return df
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--database_uri",
+        help="database URI (e.g. postgresql://ccglobal:@localhost/ccglobal)",
+        default=os.environ.get("DB_URI"),
+    )
+    parser.add_argument("--file", help="path to gzipped ghgrp file")
+    parser.add_argument(
+        "--log_file", help="path to log file", default="./ghgrp_importer.log"
+    )
+    args = parser.parse_args()
+
+    zip_file = Path(os.path.abspath(args.file))
+
+    logging.basicConfig(
+        filename=args.log_file,
+        level=logging.INFO,
+        format="%(asctime)s - %(levelname)s - %(message)s",
+    )
+
+    engine = create_engine(args.database_uri)
+    metadata_obj = MetaData()
+    Session = sessionmaker(bind=engine)
+    session = Session()
+
+    GHGRP_EPA = Table("GHGRP_EPA", metadata_obj, autoload_with=engine)
+    fields = [col.name for col in GHGRP_EPA.columns]
+
+    for year in years_to_files.keys():
+        logging.info(f"Year: {year}")
+
+        year_files = years_to_files.get(year)
+
+        for file in year_files:
+            logging.info(f"File: {file}")
+            filename = Path(file).stem
+            df = load_direct_emitters_from_zip(zip_file, file)
+            print(f'Working on: {file}')
+            
+            # preproccesing data
+            df = restructure(df)
+            df = nan_to_zero(df)
+            max_column = max_col_name(df)
+            df = assign_value_to_max_col(df, max_column)
+            df = format_change(df)
+            df['final_industry'] = df['Industry Type (sectors)'].apply(lambda x: count_words(x))
+            df = assign_gpc_ref_no(df)
+            df = add_geometry(df)
+            
+            # insert proccess
+            for _, row in df.iterrows():
+                record = row.to_dict()
+                
+                # metric tonnes to kg
+                record['emissions_quantity'] == record['emissions_quantity']*1000
+                
+                # get the locode and coordinate points
+                locode = lat_lon_to_locode(session, record['Latitude'], record['Longitude'])
+
+                # add keys filename and reno to record
+                record["locode"] = locode
+                record["filename"] = filename
+                record["gas"] = 'CO2e'
```

Comment:
Is this correct or should it be CO2 without the e?

Reply:
the emissions data from this specific source is already in CO2e, they provide the final number of emissions using the AR4 emission factors.